<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ./urdf/Roberta_P80_800.xacro   | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="Roberta_P80_800" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <origin rpy="3.14159265358 0 0" xyz="0 0 0.11"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1656_Base.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gomtec_dunkel">
        <color rgba="0.4 0.4 0.4 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="3.14159265358 0 0" xyz="0 0 0.11"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1656_Base.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <link name="link_1">
    <visual>
      <origin rpy="3.14159265358 0 3.14159265358" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1603_Joint110.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gomtec_grau">
        <color rgba="0.6 0.6 0.5 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="3.14159265358 0 3.14159265358" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1603_Joint110.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>

  <link name="link_2">
    <visual>
      <origin rpy="-3.14159265358 -1.57079632679 1.57079632679" xyz="0 0.135 0.01"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1649_Arm1.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gomtec_grau">
        <color rgba="0.6 0.6 0.5 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-3.14159265358 -1.57079632679 1.57079632679" xyz="0 0.135 0.01"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1649_Arm1.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_3">
    <visual>
      <origin rpy="-1.57079632679 0 0" xyz="0 0.1 -0.02"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1590_Joint90.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gomtec_grau">
        <color rgba="0.6 0.6 0.5 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-1.57079632679 0 0" xyz="0 0.1 -0.02"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1590_Joint90.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_4">
    <visual>
      <origin rpy="0 -1.57079632679 0" xyz="0.25 -0.0115 -0.005"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1650_Arm2.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gomtec_grau">
        <color rgba="0.6 0.6 0.5 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 -1.57079632679 0" xyz="0.25 -0.0115 -0.005"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1650_Arm2.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_5">
    <visual>
      <origin rpy="-1.57079632679 0 -3.14159265358" xyz="0 0 0.007"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1631_Joint70.STL" scale="0.001 0.001 0.001"/>
      </geometry>
     <material name="gomtec_grau">
        <color rgba="0.6 0.6 0.5 1" />
      </material>
    </visual>
    <collision>
      <origin rpy="-1.57079632679 0 -3.14159265358" xyz="0 0 0.007"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1631_Joint70.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_flange"/>
  <joint name="joint_1" type="continuous">
    <parent link="base_link"/>
    <child link="link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.185"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" velocity="1.74532925199"/>
  </joint>
  <joint name="joint_2" type="continuous">
    <parent link="link_1"/>
    <child link="link_2"/>
    <origin rpy="0 0 0" xyz="0 0 0.45"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" velocity="1.74532925199"/>
  </joint>
  <joint name="joint_3" type="continuous">
    <parent link="link_2"/>
    <child link="link_3"/>
    <origin rpy="0 0 0" xyz="0 0 0.38"/>
    <axis xyz="0 -1 0"/>
    <limit effort="0" velocity="1.74532925199"/>
  </joint>
  <joint name="joint_4" type="continuous">
    <parent link="link_3"/>
    <child link="link_4"/>
    <origin rpy="0 0 0" xyz="0 0.025 -0.01"/>
    <axis xyz="1 0 0"/>
    <limit effort="0" velocity="1.74532925199"/>
  </joint>
  <joint name="joint_5" type="continuous">
    <parent link="link_4"/>
    <child link="link_5"/>
    <origin rpy="0 0 0" xyz="0.30 0.005 -0.006"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" velocity="3.49065850399"/>
  </joint>
  <joint name="joint_6" type="continuous">
    <parent link="link_5"/>
    <child link="link_flange"/>
    <origin rpy="0 0 0" xyz="0.07 -0.002 0.000"/>
    <axis xyz="1 0 0"/>
    <limit effort="0" velocity="3.49065850399"/>
  </joint>
  <gazebo reference="base_link">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <gazebo reference="link_1">
    <material>Gazebo/Red</material>
  </gazebo>
  <gazebo reference="link_2">
    <material>Gazebo/Blue</material>
  </gazebo>
  <gazebo reference="link_3">
    <material>Gazebo/Red</material>
  </gazebo>
  <gazebo reference="link_4">
    <material>Gazebo/Blue</material>
  </gazebo>
  <gazebo reference="link_5">
    <material>Gazebo/Red</material>
  </gazebo>
  <gazebo reference="link_flange">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <link name="link_RoboCommander">
    <visual>
      <origin rpy="3.14159265359 1.57079632679 0" xyz="0 0 -0.065"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1720_RoboCommander.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gomtec_dunkel">
        <color rgba="0.4 0.4 0.4 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="3.14159265359 1.57079632679 0" xyz="0 0 -0.065"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1720_RoboCommander.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_RoboCommander_fixed" type="fixed">
    <parent link="link_flange"/>
    <child link="link_RoboCommander"/>
    <origin rpy="0 0 0" xyz="0 0 0.065"/>
  </joint>
  <gazebo reference="link_RoboCommander">
    <material>Gazebo/Green</material>
  </gazebo>

                          <!-- start of  cameralink list-->

 
 <joint name="joint" type="fixed">
        <xacro:insert_block name="origin" />
        <parent link="link_RoboCommander"/>
        <child link="cameralink"/>
      </joint>
      
      <link name="cameralink">
        <xacro:inertial_sphere mass="0.01" diameter="0.07" />
        <visual>
          <origin xyz="0 0 0" rpy="0 0 0" />
          <geometry>
            <mesh filename="package://gomtec_robot_description/meshes/visual/kinect_camera_simple.dae"/>
          </geometry>
        </visual>
        <collision>
          <origin xyz="0 0 0" rpy="0 0 0" />
          <geometry>
            <mesh filename="package://gomtec_robot_description/meshes/collision/kinect_camera_simple.stl"/>
          </geometry>
        </collision>
      </link>

      <joint name="depth_joint" type="fixed">
        <origin xyz="0.0 -0.02 0.0" rpy="0 0 0" />
        <parent link="cameralink" />
        <child link="depth_frame"/>
      </joint>

      <link name="depth_frame"/>

      <joint name="depth_optical_joint" type="fixed">
        <origin xyz="0 0 0" rpy="-1.57079632679 0.0 -1.57079632679" />
        <parent link="depth_frame" />
        <child link="depth_optical_frame"/>
      </joint>

      <link name="depth_optical_frame"/>

      <joint name="rgb_joint" type="fixed">
        <origin xyz="0.0 -0.0125 0.0" rpy="0 0 0" />
        <parent link="cameralink" />
        <child link="rgb_frame"/>
      </joint>

      <link name="rgb_frame"/>

      <joint name="rgb_optical_joint" type="fixed">
        <origin xyz="0 0 0" rpy="-1.57079632679 0.0 -1.57079632679" />
        <parent link="rgb_frame" />
        <child link="rgb_optical_frame"/>
      </joint>

      <link name="rgb_optical_frame"/>
  
      <gazebo reference="depth_frame">
        <sensor type="depth" name="depth_frame">
          <update_rate>20</update_rate>
          <camera>
            <horizontal_fov>${60 * pi/180.0}</horizontal_fov>
            <image>
              <format>B8G8R8</format>
              <width>640</width>
              <height>480</height>
            </image>
            <clip>
              <near>0.05</near>
              <far>3</far>
            </clip>
          </camera>
          <plugin name="camera_controller" filename="libgazebo_ros_openni_kinect.so">
            <imageTopicName>${perfix}/rgb/image_raw</imageTopicName>
            <cameraInfoTopicName>${perfix}/rgb/camera_info</cameraInfoTopicName>
            <depthImageTopicName>${perfix}/depth/image_raw</depthImageTopicName>
            <depthImageCameraInfoTopicName>${perfix}/depth/camera_info</depthImageCameraInfoTopicName>
            <pointCloudTopicName>${perfix}/depth/points</pointCloudTopicName>
            <frameName>${perfix}_depth_optical_frame</frameName>
          </plugin>
        </sensor>
      </gazebo>

					<!--end of link list -->

  <link name="link_Gripper">
    <visual>
      <origin rpy="1.57079632679 1.57079632679 0" xyz="0 0 0.0315"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1680_Gripper.STL" scale="0.001 0.001 0.001"/>
      </geometry>
       <material name="gomtec_grau">
        <color rgba="0.6 0.6 0.5 1" />
      </material>
    </visual>
    <collision>
      <origin rpy="1.57079632679 1.57079632679 0" xyz="0 0 0.0315"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1680_Gripper.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_Finger1">
    <visual>
      <origin rpy="1.57079632679 1.57079632679 0" xyz="-0.00323 0 0"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1690_Finger.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gomtec_gelb">
        <color rgba="1 0.7 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.57079632679 1.57079632679 0" xyz="-0.00323 0 0"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1690_Finger.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="link_Finger2">
    <visual>
      <origin rpy="1.57079632679 -1.57079632679 -3.14159265359" xyz="0.00323 0 0"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/visual/RGM1690_Finger.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gomtec_gelb">
        <color rgba="1 0.7 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.57079632679 -1.57079632679 -3.14159265359" xyz="0.00323 0 0"/>
      <geometry>
        <mesh filename="package://gomtec_robot_description/meshes/collision/RGM1690_Finger.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
             
  <link name="link_TCP"/>
  <joint name="joint_GripperBase" type="fixed">
    <parent link="link_RoboCommander"/>
    <child link="link_Gripper"/>
    <origin rpy="0 0 0" xyz="0.1 0 -0.1"/>
  </joint>
  <joint name="joint_f1_linear" type="prismatic">
    <parent link="link_Gripper"/>
    <child link="link_Finger1"/>
    <origin rpy="0 0 0" xyz="0.1 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" lower="0" upper="0.035" velocity="0.05"/>
  </joint>
  <joint name="joint_f2_linear" type="prismatic">
    <parent link="link_Gripper"/>
    <child link="link_Finger2"/>
    <origin rpy="0 0 0" xyz="0.1 0 0.07"/>
    <axis xyz="0 0 -1"/>
    <limit effort="0" lower="0" upper="0.035" velocity="0.05"/>
    <mimic joint="joint_f1_linear"/>
  </joint>
  <joint name="joint_tcp" type="fixed">
    <parent link="link_Gripper"/>
    <child link="link_TCP"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <gazebo reference="link_Gripper">
    <material>Gazebo/Green</material>
  </gazebo>
  <gazebo reference="link_Finger1">
    <material>Gazebo/Blue</material>
  </gazebo>
  <gazebo reference="link_Finger2">
    <material>Gazebo/Blue</material>
  </gazebo>
  <!--include filename="$(find gomtec_robot_description)/urdf/CameraGripper_macro.xacro"/>
	<xacro:CameraGripper prefix="" flange_link="link_RoboCommander"/-->
</robot>
